components/calendar/src/calendar/calendar-table-cell.js

```diff
@@ -0,0 +1,96 @@
+import { colors, spacers } from '@dhis2/ui-constants'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { StyleOptionsProps } from './calendar-prop-types.js'
+
+export const CalendarTableCell = ({ day, cellSize, width, styleOptions }) => {
+    return (
+        <>
+            <td data-test={day?.calendarDate} onClick={day.onClick}>
+                <button
+                    name="day"
+                    className={cx('day', {
+                        isSelected: day.isSelected,
+                        isToday: day.isToday,
+                        otherMonth: !day.isInCurrentMonth,
+                    })}
+                >
+                    {day.label}
+                </button>
+            </td>
+            <style jsx>{`
+                .calendarTable td {
+                    width: ${cellSize};
+                    height: 100%;
+                    height: ${cellSize};
+                    text-align: center;
+                }
+                .calendarTable th,
+                .calendarTable td,
+                .calendarTable tr {
+                    border: 0;
+                }
+                .calendarTable td span {
+                    display: inline-flex;
+                    padding: 2px 2px;
+                    justify-content: center;
+                    border: 2px solid transparent;
+                    min-width: 16px;
+                }
+                .selectedDate {
+                    width: ${width};
+                    padding: 4px 8px;
+                }
+                .selectedDate span {
+                    justify-content: center;
+                    font-size: 0.8em;
+                }
+                button {
+                    background: none;
+                    border: 0;
+                }
+
+                button[name='day'] {
+                    overflow: hidden;
+                    width: ${cellSize};
+                    height: ${cellSize};
+                    border: 2px solid transparent;
+                    border-radius: 3px;
+                    background: none;
+                    margin: ${spacers.dp2};
+                    color: ${colors.grey900};
+                }
+
+                button[name='day']:hover {
+                    background-color: ${styleOptions.dayHoverBackgroundColor};
+                    text-decoration: underline;
+                }
+                button[name='day'].isSelected {
+                    background-color: ${styleOptions.selectedDayBackgroundColor};
+                    color: white;
+                }
+                button.isToday {
+                    text-decoration: underline;
+                }
+                button.otherMonth {
+                    color: ${colors.grey600};
+                }
+            `}</style>
```

Comment:
These styles look off to me, considering styled-jsx will produce an encapsulated stylesheet. 

This component only contains a `td` without a class name and a `button` which can have 3 dynamic classes. So any style rules that don't apply to these 2 tags or 3 dynamic classes are out of place here (and also they won't have any effect). 

Furthermore, some of the selectors being used are much more specific than they need to be, considering this is an encapsulated stylesheet. No need to use things like `button[name=day]`, there is only one button tag in this component so just use `button`.

Finally, you don't have to render a Fragment (`<>...</>`), you can just nest the `<style jsx>...</style>` under the `td`.